0008-atomic-registry.md

```diff
@@ -0,0 +1,144 @@
+
+
+# Example
+[example]: #example
+
+In this example, Katello's registry is at katello.example.com:5001 and Atomic Registry is at katello.example.com:5000.
+
+1. An image is synced into a Katello product repository. The synced image is available directly from Katello via its contextual name which includes the organization, product, and source registry.
+
+   ```shell
+   katello.example.com:5001/examplecorp-docker_io-thomasmckay_hammer
+   ```
+   :sparkles: The project and imagestream names may be set at repository creation time. If this name is later changed, the existing project in Atomic Registry will remain.   
+   :question: If this name is later changed, the existing project in Atomic Registry will remain.  
+   :question: If a project/imagestream already exists in Atomic Registry, it will have all (or some?) tags overwritten.  
+   :question: Deleting a repository in Katello will leave the project/imagestream in Atomic Registry.
+   
+1. At repository creation, a project is created in Atomic Registry to match the namespace and an image stream is added with the original image name. The created imagestream will reference Katello's contextual image name.
+
+   ```
+   Name: hammer
+   Project: thomasmckay
+   Pull from: katello.example.com:5001/examplecorp-docker_io-thomasmckay_hammer
+   ```
+   :sparkles: Pushing to this project will be disabled to normal users as changes synced into Katello that are then automatically pushed to Atomic Registry has the potential to overwrite changes made there.
+
+1. Users may then pull images from Atomic Registry to work with. Changes should be pushed back to their own projects. This user project may then be synced into Katello for use there. In turn, the product repository in Katello may itself push back to Atomic Registry.
+   ```
+   Katello                                Atomic Registry
+   -------                                ---------------
+   *Image synced from docker.io*
+                                          *Image created with original name*
+   examplecorp-docker_io-
+        thomasmckay_hammer:latest     ->  thomasmckay/hammer:latest
+                                          *Modify, tag, then push image*
+                                          # docker tag thomasmckay/hammer:latest sallydev/hammer:latest
+                                          # docker push sallydev/hammer
+   *Image synced from Atomic Registry*
+   examplecorp-registry_example_com-  <-  
+        sallydev_hammer:latest
+                                          *Image created with new tag*
+                                      ->  thomasmckay/hammer:v1
+   ```
+
+   In this way, a user may pull an image (eg. thomasmckay/hammer:latest), manipulate it, then make it available with a new tag (eg. thomasmckay/hammer:v1). More than one Katello repo may push to a single project/imagestream, each supplying their own tags.  
```

Comment:
What is not clear from this to me is how sallydev winds up as thomasmckay/hammer:v1